SANDBOX-555: UserSignup test enhancement

rsoaresd opened the pull request:
# Description

- `TestUserResourcesCreatedWhenUserIDIsSet` and `TestUserIDAndAccountIDClaimsPropagated` does the same as `TestUserResourcesUpdatedWhenPropagatedClaimsModified`. So, we can simply delete `TestUserResourcesCreatedWhenUserIDIsSet` and `TestUserIDAndAccountIDClaimsPropagated` to avoid duplicated code.
- `TestProvisionToOtherClusterWhenOneIsFull` does almost the same as `TestAutomaticApproval`. The only "extra verifications" are a few assertions. So, we can merge them to clean up the code.
- Deleted `doc.go` file since it is empty.
- Tests `TestTransformUsernameWithSpaceConflict` and  `TestTransformUsername` can be run in parallel.


With the deleted tests, we can save nearly 2 mins:
| Test                                     | Execution Time (s)         |
|------------------------------------------|------------------|
| TestUserResourcesCreatedWhenUserIDIsSet  | ~42              |
| TestUserIDAndAccountIDClaimsPropagated  | ~41              |
| TestProvisionToOtherClusterWhenOneIsFull | ~49              |
| **Total**                                | **~132 (~2mins)** |

With the tests that will run in parallel, we will save time too since their execution time is:
| Test                                   | Execution Time (s)           |
|----------------------------------------|--------------------|
| TestTransformUsername                  | ~177               |
| TestTransformUsernameWithSpaceConflict | ~47                |
| **Total**                              | **~224 (~3/4min)** |


## Issue ticket number and link
[SANDBOX-555](https://issues.redhat.com/browse/SANDBOX-555)

alexeykazakov commented on test/e2e/usersignup_test.go:
```diff
@@ -72,11 +67,18 @@ func (s *userSignupIntegrationTest) TestAutomaticApproval() {
-			Execute(s.T())
+			Execute(s.T()).Resources()
+
+		require.NotEqual(t, mur1.Status.UserAccounts[0].Cluster.Name, mur2.Status.UserAccounts[0].Cluster.Name)
```
Could you please also update the test description when you are merging tests?
For example in this case the test description is saying that it testing that when the cluster is full then new spaces are not approved but put in the wait list. However it also now tests then when one cluster if full then a new space is places in another cluster.

Just keep in mind that test execution time is one thing but we also should keep the test readable/maintainable as much as possible. And adding proper comments in the code can help with that.

alexeykazakov reviewed (commented):
Thank you for improving the tests! I just started reviewing it but wanted to ask to add more comments to the updated tests so it's clear what they are doing now.

alexeykazakov commented:
/ok-to-test

MatousJobanek commented:
/ok-to-test

alexeykazakov commented on test/e2e/parallel/usersignup_test.go:
```diff
@@ -0,0 +1,214 @@
+	}
+}
+
+func newSignupRequestWithManualApprove(t *testing.T, awaitilities wait.Awaitilities, testCase TestCase) {
```
Or maybe something like this would be even more accurate:
```suggestion
func assertComplaintUsernameForNewSignup(t *testing.T, awaitilities wait.Awaitilities, testCase TestCase) {
```

alexeykazakov commented on test/e2e/parallel/usersignup_test.go:
```diff
@@ -0,0 +1,214 @@
+type TestCase struct {
+	username          string
+	email             string
+	compliantUsername string
```
```suggestion
	expectedCompliantUsername string
```

alexeykazakov commented on test/e2e/usersignup_test.go:
```diff
@@ -355,10 +265,16 @@ func (s *userSignupIntegrationTest) TestUserResourcesUpdatedWhenPropagatedClaims
 		Username("test-user-resources-updated").
 		Email("test-user-resources-updated@redhat.com").
 		UserID("43215432").
+		// 	AccountID("987654321").
```
```suggestion
```

alexeykazakov reviewed (commented):
Looks good! Just a few cosmetic suggestions.

alexeykazakov reviewed (approved):
Thank you for addressing my comments! But please do not push with force or squash existing commits in the PR because it's harder to do followup reviews. When there are new commits added to address review comments then we don't have to review everything from scratch and can review the changes since the last review.
No worries about too many commits in the PR. They will all be squashed when we merge them to master.

rsoaresd commented:
> Thank you for addressing my comments! But please do not push with force or squash existing commits in the PR because it's harder to do followup reviews. When there are new commits added to address review comments then we don't have to review everything from scratch and can review the changes since the last review. No worries about too many commits in the PR. They will all be squashed when we merge them to master.

Deal! Thank you so much!

rajivnathan reviewed (approved):
Cool, looks good. Thanks for the improvement! 👍

alexeykazakov commented:
/lgtm
